src/auction.js

```diff
@@ -514,28 +514,67 @@ function tryAddVideoBid(auctionInstance, bidResponse, afterBidAdded, {index = au
   }
 }
 
-export const callPrebidCache = hook('async', function(auctionInstance, bidResponse, afterBidAdded, videoMediaType) {
-  store([bidResponse], function (error, cacheIds) {
-    if (error) {
-      logWarn(`Failed to save to the video cache: ${error}. Video bid must be discarded.`);
-
-      doCallbacksIfTimedout(auctionInstance, bidResponse);
-    } else {
-      if (cacheIds[0].uuid === '') {
-        logWarn(`Supplied video cache key was already in use by Prebid Cache; caching attempt was rejected. Video bid must be discarded.`);
+const storeInCache = (batch) => {
+  store(batch.map(entry => entry.bidResponse), function (error, cacheIds) {
+    cacheIds.forEach((cacheId, i) => {
+      const { auctionInstance, bidResponse, afterBidAdded } = batch[i];
+      if (error) {
+        logWarn(`Failed to save to the video cache: ${error}. Video bid must be discarded.`);
 
         doCallbacksIfTimedout(auctionInstance, bidResponse);
       } else {
-        bidResponse.videoCacheKey = cacheIds[0].uuid;
+        if (cacheId.uuid === '') {
+          logWarn(`Supplied video cache key was already in use by Prebid Cache; caching attempt was rejected. Video bid must be discarded.`);
 
-        if (!bidResponse.vastUrl) {
-          bidResponse.vastUrl = getCacheUrl(bidResponse.videoCacheKey);
+          doCallbacksIfTimedout(auctionInstance, bidResponse);
+        } else {
+          bidResponse.videoCacheKey = cacheId.uuid;
+
+          if (!bidResponse.vastUrl) {
+            bidResponse.vastUrl = getCacheUrl(bidResponse.videoCacheKey);
+          }
+          addBidToAuction(auctionInstance, bidResponse);
+          afterBidAdded();
         }
-        addBidToAuction(auctionInstance, bidResponse);
-        afterBidAdded();
       }
-    }
+    });
   });
+};
+
+const batchAndStore = (() => {
```

Comment:
A trick to make testing easier is to inject `setTimeout` and `storeInCache` as dependencies here, e.g.

```javascript
export function batchingCache(timeout = setTimeout, cache = storeInCache) {
    ...
    timeout(() => {
        batches.forEach(cache);
        ....
    })
}
const batchAndStore = batchingCache();
```

so that you can write (or reuse) tests for `storeInCache` without worrying about batching, and then inject stubs here to test batching without worrying about `storeInCache`:

```
let stubCache = sinon.stub();
let mockTimeout = sinon.stub().callsFake((cb) => cb())
const batchAndStore = batchingCache(mockTimeout, stubCache);
...
sinon.assert.calledWith(mockTimeout, sinon.match.any(), expectedBatchTimeout);
sinon.assert.calledWith(stubCache, expectedBatch);
```